include/swift/AST/ASTPrinter.h

```diff
@@ -263,6 +263,14 @@ class ASTPrinter {
     printStructurePre(Kind, D);
   }
 
+  /// Make a callback to printStructurePost(), performing any necessary
+  /// bookkeeping.
+  void callPrintStructurePost(PrintStructureKind Kind,
+                              const Decl *D = nullptr) {
+    forceNewlines();
```

Comment:
How do we get in a situation where we want to print newlines before the "post" callback?  Usually we force newlines before the "pre" so that the callback happens at the point where the contents is being inserted.  For "post" this seems backwards to me.